pkgs/development/python-modules/speechrecognition/default.nix

```diff
@@ -26,29 +32,43 @@ buildPythonPackage rec {
     hash = "sha256-icXZUg2lVLo8Z5t9ptDj67BjQLnEgrG8geYZ/lZeJt4=";
   };
 
+  postPatch = ''
+    # Remove Bundled binaries
+    rm speech_recognition/flac-*
+    rm -r third-party
+
+    substituteInPlace speech_recognition/audio.py \
+      --replace-fail 'shutil_which("flac")' '"${lib.getExe flac}"'
+  '';
+
   build-system = [ setuptools ];
 
   dependencies = [
+    pyaudio
     requests
     typing-extensions
   ];
 
-  nativeCheckInputs = [
-    numpy
-    pytestCheckHook
-    torch
-    soundfile
-  ];
+  optional-dependencies = {
+    amazone = [ boto3 ];
+    google-cloud-speech = [ google-cloud-speech ];
+    lex = [ boto3 ];
+    pocketsphinx = [ pocketsphinx ];
+    tensorflow = [ tensorflow ];
```

Comment:
Where are these dependencies defined?

Reply:
They sadly do not list in the dependencies list but can be found in the backend recognizer in https://github.com/Uberi/speech_recognition/blob/master/speech_recognition/__init__.py

Reply:
optional-dependencies are really reserved for extras https://github.com/Uberi/speech_recognition/blob/master/setup.cfg#L6-L14

Reply:
Okay, makes sense to me.